File: archive/ruby/muse/algorithms/sorting/InsertionSort.rb
module InsertionSort
  def self.sort(arr)
    target = nil

    cursor = 0

    1.upto(arr.length - 1) do |i|
      target = arr[i]
      cursor = i
      while cursor.positive?
        break if arr[cursor - 1] <= target

        arr[cursor] = arr[cursor - 1]
        cursor -= 1
      end
      arr[cursor] = target
    end
  end
end

File: archive/php/muse/algorithms/sorting/InsertionSort.php
<?php

declare(strict_types=1);

namespace muse\algorithms\sorting\InsertionSort;

function sort(array &$arr): void
{
    $target = null;

    $cursor = 0;

    for ($i = 1, $length = count($arr); $i < $length; $i++) {
        $target = $arr[$i];
        for ($cursor = $i; $cursor > 0; $cursor--) {
            if ($arr[$cursor - 1] <= $target) {
                break;
            }
            $arr[$cursor] = $arr[$cursor - 1];
        }
        $arr[$cursor] = $target;
    }
}
